dvc/repo/metrics/show.py

```diff
@@ -171,20 +172,28 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
             - xpath:
     """
     outs = [out for stage in repo.stages() for out in stage.outs]
+    found = []
 
-    if path:
+    # Display all metrics if no target is given
+    if not targets:
+        targets = []
+        found = outs
+
+    for target in targets:
```

Comment:
This feels weird, this `for` is disconnected from `if not targets` above. How about something like
```
if targets:
    # maybe even move
    for target in targets:
    ....
else:
    found = outs
```